tokio/src/io/registration.rs

```diff
@@ -63,12 +63,34 @@ impl Registration {
     /// from a future driven by a tokio runtime, otherwise runtime can be set
     /// explicitly with [`Handle::enter`](crate::runtime::Handle::enter) function.
     pub fn new<T>(io: &T) -> io::Result<Registration>
+    where
+        T: Evented,
+    {
+        Registration::new_with_ready(io, mio::Ready::all())
+    }
+
+    /// Registers the I/O resource with the default reactor, for a specific `mio::Ready` state.
```

Comment:
It may be worth noting here that this will not register HUP interest. `Ready::all()` also registers interest in errors, but those should trigger without registration anyways.

Reply:
Should we combine the readiness they provide with hup and error?

Reply:
My concern about that is that if a user *really* does not want to register HUP interest for example, then `new_with_ready` seems like the method that should be used. It would be surprising if we still automatically register that.

Reply:
I've added additional documentation to PollEvented and Registration. Let me know if that clarifies things.

Reply:
Yep this looks good.